3 tests from the TestQueueWithStacks class are failing

...as part of an exercise I did with @tshur 
```
$ pytest --doctest-modules --cov=dsa --cov-report xml:coverage.xml
====================================================================================================== test session starts =======================================================================================================
platform darwin -- Python 3.13.0, pytest-8.3.5, pluggy-1.6.0
rootdir: /Users/dan/code/tshur/py-alg
configfile: pyproject.toml
plugins: testmon-2.1.3, cov-6.1.1
collected 335 items                                                                                                                                                                                                              

dsa/graph/graph_test.py ........................                                                                                                                                                                           [  7%]
dsa/graph/matrix_graph.py ...........                                                                                                                                                                                      [ 10%]
dsa/graph/node_graph.py ..........                                                                                                                                                                                         [ 13%]
dsa/hash/map_linked_list.py ......                                                                                                                                                                                         [ 15%]
dsa/hash/map_list.py ......                                                                                                                                                                                                [ 17%]
dsa/hash/map_test.py ..................                                                                                                                                                                                    [ 22%]
dsa/hash/set.py .......                                                                                                                                                                                                    [ 24%]
dsa/hash/set_test.py .......                                                                                                                                                                                               [ 26%]
dsa/heap/heap.py ............                                                                                                                                                                                              [ 30%]
dsa/heap/heap_test.py ............                                                                                                                                                                                         [ 33%]
dsa/iterable/adjacent_transform.py ..                                                                                                                                                                                      [ 34%]
dsa/iterable/adjacent_transform_test.py .........                                                                                                                                                                          [ 37%]
dsa/iterable/reduce.py .                                                                                                                                                                                                   [ 37%]
dsa/iterable/reduce_test.py .....                                                                                                                                                                                          [ 38%]
dsa/iterable/reverse.py .                                                                                                                                                                                                  [ 39%]
dsa/iterable/reverse_test.py ......                                                                                                                                                                                        [ 40%]
dsa/iterable/rotate.py .                                                                                                                                                                                                   [ 41%]
dsa/iterable/rotate_test.py .....                                                                                                                                                                                          [ 42%]
dsa/iterable/scan.py .                                                                                                                                                                                                     [ 42%]
dsa/iterable/scan_test.py .....                                                                                                                                                                                            [ 44%]
dsa/iterable/skip.py .                                                                                                                                                                                                     [ 44%]
dsa/iterable/skip_test.py .....                                                                                                                                                                                            [ 46%]
dsa/linked_list/doubly_linked_list.py .....                                                                                                                                                                                [ 47%]
dsa/linked_list/linked_list_test.py ......................                                                                                                                                                                 [ 54%]
dsa/linked_list/singly_linked_list.py .....                                                                                                                                                                                [ 55%]
dsa/queue/deque.py ..............                                                                                                                                                                                          [ 60%]
dsa/queue/deque_test.py ............                                                                                                                                                                                       [ 63%]
dsa/queue/queue.py .........                                                                                                                                                                                               [ 66%]
dsa/queue/queue_test.py .......                                                                                                                                                                                            [ 68%]
dsa/search/binary_search.py ...                                                                                                                                                                                            [ 69%]
dsa/search/binary_search_test.py .............                                                                                                                                                                             [ 73%]
dsa/search/linear_search.py .                                                                                                                                                                                              [ 73%]
dsa/search/linear_search_test.py .....                                                                                                                                                                                     [ 74%]
dsa/sort/heap_sort.py .                                                                                                                                                                                                    [ 75%]
dsa/sort/insertion_sort.py .                                                                                                                                                                                               [ 75%]
dsa/sort/selection_sort.py .                                                                                                                                                                                               [ 75%]
dsa/sort/sort_test.py ....................                                                                                                                                                                                 [ 81%]
dsa/sort/tree_sort.py .                                                                                                                                                                                                    [ 82%]
dsa/stack/stack.py ........                                                                                                                                                                                                [ 84%]
dsa/stack/stack_test.py .......                                                                                                                                                                                            [ 86%]
dsa/tree/binary_search_tree.py ..........                                                                                                                                                                                  [ 89%]
dsa/tree/binary_search_tree_test.py .........                                                                                                                                                                              [ 92%]
leetcode/linked_list/detect_cycle_test.py ........                                                                                                                                                                         [ 94%]
leetcode/linked_list/reverse_linked_list_test.py ............                                                                                                                                                              [ 98%]
leetcode/stack/queue_with_stacks_test.py .F.F.F                                                                                                                                                                            [100%]

============================================================================================================ FAILURES ============================================================================================================
_____________________________________________________________________________________________ TestQueueWithStacks.test_from_iterable _____________________________________________________________________________________________

self = <stack.queue_with_stacks_test.TestQueueWithStacks object at 0x10b4ed310>

    def test_from_iterable(self) -> None:
        queue = QueueWithStacks.from_iterable([1, 2, 3, 4, 5])
    
        assert len(queue) == 5
        assert not queue.is_empty()
    
>       assert queue.peek() == 1
E       assert 5 == 1
E        +  where 5 = peek()
E        +    where peek = <stack.queue_with_stacks.QueueWithStacks object at 0x10b4efd90>.peek

leetcode/stack/queue_with_stacks_test.py:23: AssertionError
________________________________________________________________________________________________ TestQueueWithStacks.test_remove _________________________________________________________________________________________________

self = <stack.queue_with_stacks_test.TestQueueWithStacks object at 0x10b4b2190>

    def test_remove(self) -> None:
        queue = QueueWithStacks.from_iterable([10, 20, 30])
    
        assert len(queue) == 3
>       assert queue.peek() == 10
E       assert 30 == 10
E        +  where 30 = peek()
E        +    where peek = <stack.queue_with_stacks.QueueWithStacks object at 0x10b4b3490>.peek

leetcode/stack/queue_with_stacks_test.py:44: AssertionError
_________________________________________________________________________________________ TestQueueWithStacks.test_alternate_add_remove __________________________________________________________________________________________

self = <stack.queue_with_stacks_test.TestQueueWithStacks object at 0x10b32b680>

    def test_alternate_add_remove(self) -> None:
        queue = QueueWithStacks.from_iterable([1, 2, 3])
    
>       assert queue.dequeue() == 1
E       assert 3 == 1
E        +  where 3 = dequeue()
E        +    where dequeue = <stack.queue_with_stacks.QueueWithStacks object at 0x10b5669c0>.dequeue

leetcode/stack/queue_with_stacks_test.py:66: AssertionError
========================================================================================================= tests coverage =========================================================================================================
________________________________________________________________________________________ coverage: platform darwin, python 3.13.0-final-0 ________________________________________________________________________________________

Coverage XML written to file coverage.xml
==================================================================================================== short test summary info =====================================================================================================
FAILED leetcode/stack/queue_with_stacks_test.py::TestQueueWithStacks::test_from_iterable - assert 5 == 1
FAILED leetcode/stack/queue_with_stacks_test.py::TestQueueWithStacks::test_remove - assert 30 == 10
FAILED leetcode/stack/queue_with_stacks_test.py::TestQueueWithStacks::test_alternate_add_remove - assert 3 == 1
================================================================================================= 3 failed, 332 passed in 6.00s ==================================================================================================
```

--- a/leetcode/stack/queue_with_stacks.py
+++ b/leetcode/stack/queue_with_stacks.py
@@ -93,8 +93,8 @@ class QueueWithStacks:
         while self._in_stack:
             element = self._in_stack.pop()
 
-            # this if statement seems redundant but its required to avoid a type checker
-            # warning for the push arg
+            # This if statement is redundant since the popped element should always
+            # exist here. Kept to avoid a type checker warning.
             if element is not None:
                 self._out_stack.push(element)
 

